### PR TITLE
Xenial rust autoupdate

### PR DESCRIPTION
**Pull request by TriplEight**
- should autoupdate when `rust:slim` does,
- `ubuntu:xenial` based,
- with `sccache` which will allow to proceed with caching between jobs,
- new `label`s to better understand what happens.

Tested, 
- able to build, 
- able to launch in virgin `ubuntu:xenial`.

offer to use this container before we will transfer to static linker.

### CODE REVIEW

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,27 @@
-FROM rust:stretch
-LABEL maintainer="Parity Technologies <devops-team@parity.io>"
-LABEL description="This is for CI build stage for Parity Ethereum linux binary."
+# triggers image autoupdate with the new Rust release
+FROM rust:slim as notifier
```
Comment (5chdn) [inline docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile]: that's pretty cool :+1:

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
 RUN apt-get -y update && \
 	apt-get install -y --no-install-recommends \
 	software-properties-common curl git \
 	make cmake ca-certificates g++ rhash \
-	gcc pkg-config libudev-dev time
-
-# removed:
-# binutils binutils-dev snapcraft gettext file python build-essential zip dpkg-dev rpm libssl-dev openssl ruby-dev
+	gcc pkg-config libudev-dev time \
+	libssl-dev
```
Comment (5chdn) [inline docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile]: why did you add libssl back in?

Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile]: unfortunately, it's required for installations of `sccache` and `audit` in `ubuntu:xenial`, and in `debian:stretch` it's just built-in.

File: docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile (hunk 1 above)
Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci/Dockerfile]: and weighs not that much, but I consider it a crutch though xD
